### PR TITLE
Antoinette/swipe all

### PR DESCRIPTION
**Pull request by antoinztte**
## Overview

- applied push EateryPageVC to list of eateries

## Changes Made

- modified pushViewController method to push EateryPageVC instead of HomeModel
- modified DidSelectRowAt

## Test Coverage

- tested on iPhone 14 all sizes

## Next Steps (delete if not applicable)

- fix indexing (based on Jennifer's code once approved) and also figure out how to show all pages at once

## Related PRs or Issues (delete if not applicable)

- Jennifer's recent swipe index PR

### CODE REVIEW

File: EateryModel/Sources/EateryModel/Networking/SchemaToModel.swift [hunk 1]
```diff
@@ -76,9 +83,18 @@ internal enum SchemaToModel {
     }
 
     internal static func convert(_ schemaEvent: Schema.Event) -> Event? {
-        guard let day = Day(string: schemaEvent.canonicalDate) else {
+        let df = ISO8601DateFormatter()
```
Comment (ReadeP) [inline EateryModel/Sources/EateryModel/Networking/SchemaToModel.swift:86]: rename to `dateFormatter`

**Review by ReadeP — APPROVED** [review]
Looks good! 👍

File: Eatery Blue/Eatery Blue/UI/HomeScreen/HomeViewController.swift [hunk 2]
```diff
@@ -282,11 +283,12 @@ extension HomeViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         switch cells[indexPath.row] {
         case .eateryCard(eatery: let eatery):
-            pushViewController(for: eatery)
-
+            allEateries.append(eatery)
```
Comment (jjennifergu) [inline Eatery Blue/Eatery Blue/UI/HomeScreen/HomeViewController.swift:286]: if you append each eatery in `didSelectRowAt`, the order of the eateries in `allEateries` will be in the order you click on them, which will not be the order of the eateries on the screen. also, you will only be able to scroll through the eateries you have previously clicked on, which is not what we want in this case. I would suggest getting `allEateries` from `HomeModelController`, where the data is fetched from backend

File: EateryModel/Sources/EateryModel/Networking/EateryAPI.swift [hunk 3]
```diff
@@ -22,13 +22,13 @@ public struct EateryAPI {
 
     public func eateries() async throws -> [Eatery] {
         let (data, _) = try await URLSession.shared.data(from: url, delegate: nil)
-        let schemaApiResponse = try decoder.decode(Schema.APIResponse.self, from: data)
+        let schemaApiResponse = try decoder.decode([Schema.Eatery].self, from: data)
 
-        if let error = schemaApiResponse.error {
-            throw EateryAPIError.apiResponseError(error)
-        }
+//        if let error = schemaApiResponse.error {
```
[the remark this answers is not in the document]
Comment (jjennifergu) [reply EateryModel/Sources/EateryModel/Networking/EateryAPI.swift:27]: this is addressed in my branch that I have already merged, i can help you resolve merge conflicts when you merge this branch. nothing needs to be done here for now

File: EateryModel/Sources/EateryModel/Networking/Schema.swift [hunk 4]
```diff
@@ -21,29 +21,27 @@ internal enum Schema {
...
-        internal let error: String?
-
-        internal let success: Bool
-
-    }
+//    internal struct APIResponse: Codable {
+//
+//        internal let data: [Eatery]
+//
+//        internal let error: String?
```
[the remark this answers is not in the document]
Comment (jjennifergu) [reply EateryModel/Sources/EateryModel/Networking/Schema.swift:28]: same as above, will help you address when you merge

File: EateryModel/Sources/EateryModel/Networking/Schema.swift [hunk 5]
```diff
@@ -63,19 +61,25 @@ internal enum Schema {
...
 
-        internal let waitTimes: [Schema.WaitTimesByDay]?
+        internal let waitTimes: [Schema.WaitTimesByDay]? // Not implemeneted on backend currently, will always be nil
 
     }
 
     internal struct Event: Codable {
 
-        internal let canonicalDate: String
+//        internal let canonicalDate: String
```
[the remark this answers is not in the document]
Comment (jjennifergu) [reply EateryModel/Sources/EateryModel/Networking/Schema.swift:70]: same as above

**Review by jjennifergu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**